packages/grid/_modules_/grid/components/GridPagination.tsx

```diff
@@ -80,6 +83,25 @@ export const GridPagination = React.forwardRef<
     };
   };
 
+  if (process.env.NODE_ENV !== 'production') {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const warnedOnceMissingPageSizeInRowsPerPageOptions = React.useRef(false);
+    if (
+      !warnedOnceMissingPageSizeInRowsPerPageOptions.current &&
+      !rootProps.rowsPerPageOptions!.includes(rootProps.pageSize ?? DEFAULT_PAGE_SIZE)
```

Comment:
We can give a default value for the prop in:

https://github.com/mui-org/material-ui-x/blob/75ae2f73204939c73ae3b8f5597b6541fefe34dc/packages/grid/data-grid/src/DataGrid.tsx#L53

The same for the XGrid.

```suggestion
      !rootProps.rowsPerPageOptions!.includes(rootProps.pageSize)
```

Reply:
If we do so, then `useGridPageSize` will always be in controlled mode no ?
We could give the default props only to the context and not to the feature hooks but I don't think it's a good idea.

Reply:
We can use the state value, If think. One test related to auto size will fail, maybe we don't display the warning in this condition because there's no way to change the rows per page.

```suggestion
      !rootProps.rowsPerPageOptions!.includes(rootProps.pageSize ?? paginationState.pageSize)
```